nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -308,7 +308,6 @@ in
         StateDirectory = mkIf (cfg.dataDir == "/var/lib/caddy") [ "caddy" ];
         LogsDirectory = mkIf (cfg.logDir == "/var/log/caddy") [ "caddy" ];
         Restart = "on-abnormal";
-        SupplementaryGroups = mkIf (length acmeVHosts != 0) [ "acme" ];
```

Comment:
Removing this would be a pretty significantly breaking change, but it can also be argued this code was too naive to begin with.

With other web servers, [a default group is set](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-servers/apache-httpd/default.nix#L689) which would ensure that certificates would be accessible by the service if minimally configured.

For all other configurations, we assume that the user will use either SupplementaryGroups or extra groups for the web server's user (cfg.user) to grant access to the certificates. This way, users are aware of when the acme group is being shared across services as they have to do so explicitly. 

I agree that we shouldn't be automatically expanding the permissions scope of a service like this, but just to close the loop I'd like to point out this code originally should have iterated the acmeVHosts and build a list of SupplementaryGroups from that, rather than hard-coding "acme" as the one and only to add. What if the user made their own group like "acme-webservers" for example. Does that make sense?

Reply:
> Removing this would be a pretty significantly breaking change

IMHO, this should not cause any break.

> [a default group is set](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/web-servers/apache-httpd/default.nix#L689)

Is only setting `group` enough? Or should I also add `webroot`, `dnsProvider` and `extraDomains` like [nginx](https://github.com/NixOS/nixpkgs/blob/2e03101f0166d24cd41e8ef38b7efea01d41159b/nixos/modules/services/web-servers/apache-httpd/default.nix#L689-L699) does?

Reply:
Currently, I only set default `group`.

Reply:
There's no need to set anything else because (at least from what I can see?) there's no similar `acmeRoot` option for caddy vhosts.